frontend/src/components/new-position-form.js

```diff
@@ -0,0 +1,151 @@
+    }
+};
+const validInstructor = (val, props) => {
+    const instructors = props.instructors.map(toNames);
+    return instructors.indexOf(val) !== -1 ? false : "Not a valid instructor";
+};
+
+export const newPositionFields = [
+    {
+        label: "Course Code",
+        value: "course_code",
+        validate: [validCourseCode],
+        required: true
+    },
+    {
+        label: "Course Name",
+        value: "course_name",
+        validate: [],
+        required: false
+    },
+    {
+        label: "Current Enrolment",
+        value: "cap_enrolment",
+        validate: [validNumber],
+        required: false
+    },
+    { label: "Duties", value: "duties", validate: [], required: false },
+    {
+        label: "Qualifications",
+        value: "qualifications",
+        validate: [],
+        required: false
+    },
+    {
+        label: "Instructor",
+        value: "instructor",
+        validate: [validInstructor],
+        required: true
+    },
+    {
+        label: "Session",
+        value: "session_id",
+        validate: [validNumber],
+        required: true
+    },
+    {
+        label: "Hours",
+        value: "hours",
+        validate: [validNumber],
+        required: false
+    },
+    {
+        label: "Openings",
+        value: "openings",
+        validate: [validNumber],
+        required: true
+    },
+    {
+        label: "Start Date",
+        value: "start_date",
+        validate: [validDate],
+        required: false
+    },
+    {
+        label: "End Date",
+        value: "end_date",
+        validate: [validDate],
+        required: false
+    }
+];
+
+export const DefaultInput = ({ label, curValue, onChange, required }) => (
+    <div>
+        <label style={{ width: "100px" }}>
+            {required ? label + "*" : label}
+        </label>
+        <input value={curValue} onChange={onChange} required={required} />
+    </div>
+);
+
+export const TextboxInput = ({ label, curValue, onChange, required }) => (
+    <div>
+        <label style={{ width: "100px" }}>
+            {required ? label + "*" : label}
+        </label>
+        <textarea
+            style={{ width: "152px" }}
+            value={curValue}
+            onChange={onChange}
+            required={required}
+        />
+    </div>
+);
+
+export const InstructorInput = ({ curValue, label, onChange, required }) => (
+    <div>
+        <label style={{ width: "100px" }}>
+            {required ? label + "*" : label}
+        </label>
+        <input value={curValue} onChange={onChange} required={required} />
+    </div>
+);
+
+export const InstructorList = ({ curValue, instructors, setInstructor }) => {
+    const valid = validInstructor(curValue, { instructors });
+    const matchingInstructors = instructors
+        .map(({ first_name, last_name }) => `${first_name} ${last_name}`)
+        .filter(
+            name => name.toLowerCase().indexOf(curValue.toLowerCase()) !== -1
+        );
```

Comment:
This confuses me...how could there me more than one matching instructor?

Reply:
![Screenshot from 2019-10-10 21-28-19](https://user-images.githubusercontent.com/10068531/66617913-1078e380-eba5-11e9-9295-15fff3714e65.png)